packages/core/src/components/forms/numericInput.tsx

```diff
@@ -479,15 +479,11 @@ export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInp
 
         this.shouldSelectAfterUpdate = false;
         this.setState({ value: nextValue });
-        this.invokeOnValueChangeCallback(nextValue);
+        this.invokeValueCallback(nextValue, this.props.onValueChange);
     }
 
-    private invokeOnValueChangeCallback(value: string) {
-        Utils.safeInvoke(this.props.onValueChange, +value, value);
-    }
-
-    private invokeOnButtonClickCallback(value: string) {
-        Utils.safeInvoke(this.props.onButtonClick, +value, value);
+    private invokeValueCallback(value: string, callback: (valueAsNumber: number, valueAsString: string) => void) {
```

Comment:
wait i don't understand why you can't set `this.props.onValueChange` as the default value for `callback` parameter? 

maybe the `invokeValueCallback` in `incrementValue` is causing those double-events?